service/websocketserver/setup.go

```diff
@@ -110,7 +110,8 @@ func (wss *WebSocketServer) handleWebSocket(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	serviceName := strings.TrimPrefix(spiffeID.Path(), "/")
+	pathParts := strings.Split(spiffeID.Path(), "/")
+	serviceName := pathParts[len(pathParts)-1]
```

Comment:
```suggestion
    var serviceName string
    if spiffeID.String() == wss.TokenetesSpiffeId.String() {
      serviceName = common.TOKENETES_SERVICE_NAME
    } else {
      serviceName = extractServiceNameFromSPIFFEPath(spiffeID.Path())
    }
```

And the function `extractServiceNameFromSPIFFEPath` can be:

```
  // extractServiceNameFromSPIFFEPath extracts the service name from a SPIFFE ID path.
  // For Kubernetes SPIFFE IDs, it looks for /sa/ pattern and extracts the service account name.
  // Falls back to the last path component if /sa/ pattern is not found.
  func extractServiceNameFromSPIFFEPath(path string) string {
      if saIndex := strings.Index(path, "/sa/"); saIndex != -1 {
          // Found /sa/ pattern, extract the service account name
          saStart := saIndex + 4 // skip "/sa/"
          pathAfterSA := path[saStart:]
          if nextSlash := strings.Index(pathAfterSA, "/"); nextSlash != -1 {
              return pathAfterSA[:nextSlash]
          }

          return pathAfterSA
      }

      // Fallback to last path component
      pathParts := strings.Split(path, "/")

      return pathParts[len(pathParts)-1]
  }
  ```